thunder/tests/test_autocast.py

```diff
@@ -169,3 +169,29 @@ def foo(x, w):
         jit_out = jfoo(x, w)
 
     torch.testing.assert_close(eager_out, jit_out)
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+def test_autocast_convolution(dim):
+    conv_fn = getattr(torch.nn.functional, f"conv{dim}d")
+
+    def foo(x, w, b=None):
+        return conv_fn(x, w, b)
+
+    x = torch.randn(1, 2, *(dim * (8,)))
+    w = torch.randn(3, 2, *(dim * (4,)))
+    b = torch.randn(3)
+
+    jfoo = thunder.jit(foo)
+
+    with torch.autocast("cpu", torch.bfloat16):
+        eager_out = foo(x, w, b)
+        jit_out = jfoo(x, w, b)
+
+    torch.testing.assert_close(eager_out, jit_out)
+
+    with torch.autocast("cpu", torch.bfloat16):
+        eager_out = foo(x, w)
+        jit_out = jfoo(x, w)
```

Comment:
I wonder, will the autograd be affected?

Reply:
you mean the backward?

Reply:
Yeah :)

Reply:
Added it with terrible bounds...

Reply:
Hmmm, maybe `rand` vs `randn` could do better?

Reply:
Well, so I would suspect that there still is some dragon lingering somewhere, because I cannot really think of  much reason to have thunder + autocast return wildly different results than torch + autocast even for wild inputs, but TBH, I would postpone investigating that.

Reply:
but you are right, switching to rand buys me an order of magnitude, cool, thank you!

Reply:
Turns out that Thunder does funny, slow things for the conv backward.

Reply:
https://github.com/Lightning-AI/lightning-thunder/issues/799 <-- filed an issue